include/hdlConvertor/hdlObjects/hdlTypeDec.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include <vector>
+#include <utility>
+#include <memory>
+#include <hdlConvertor/hdlObjects/named.h>
+#include <hdlConvertor/hdlObjects/iHdlObj.h>
+
+
+namespace hdlConvertor {
+namespace hdlObjects {
+
+class iHdlExpr;
+class HdlVariableDef;
+
+/*
+ * HDL AST node for module declaration
+ * (part with ports and params for Verilog, Entity for VHDL)
+ * */
+class HdlTypeDec: public WithNameAndDoc, public iHdlObj {
+public:
+	// enum to indicate which class of type declaration this is, useful
+	// when visiting the AST
+	enum TypeClass {hdltc_error, hdltc_incomplete, hdltc_enum, hdltc_array, 
+		hdltc_struct, hdltc_class, hdltc_physical, hdltc_integral, 
+		hdltc_float, hdltc_access, hdltc_file, hdltc_protected};
+
+	HdlTypeDec(const std::string& name);
+	virtual TypeClass typeClass() {return hdltc_error;};
```

Comment:
Repeating a previous comment, should we use the base type `HdlTypeDec` to represent the "error" type concept or "incomplete" type concept?  Right now the `typeClass` method is returning the "error" concept.  I suppose we could use a flag to distinguish, but it will end up in all of the derived classes as well.  Using the method-returning-constant approach also makes this harder.  Maybe an "other" derived type is a better solution.